File: Decorator/Java/CoffeeShowDecorator/src/CoffeeShopDecorator/SoyCondiment.java
package CoffeeShopDecorator;

//ConcreteDecorator
public class SoyCondiment extends CondimentDecorator
{
    public SoyCondiment(Beverage beverage) 
    {
    	super(beverage);
        description = "soy";
    }

    @Override
    public double getPrice()
    {
        return wrappedBeverage.getPrice() + 0.2d;
    }

    @Override
    public String getDescription()
    {
        return String.format("%s, %s", wrappedBeverage.getDescription(), description);
    }
}

File: Decorator/C#/CoffeeShopDecorator/SoyCondiment.cs
namespace CoffeeShopDecorator
{
    //ConcreteDecorator
    public class SoyCondiment : CondimentDecorator
    {
        public SoyCondiment(Beverage beverage) : base(beverage)
        {
            Description = "soy";
        }

        public override string GetDescription()
        {
            return $"{base.GetDescription()}, {Description}";
        }

        public override decimal GetPrice()
        {
            return base.GetPrice() + 0.2m;
        }
    }
}